ppb/sprites.py

```diff
@@ -19,6 +26,113 @@
 side_attribute_error_message = error_message.format
 
 
+class BaseSprite(EventMixin):
+    """
+    The base Sprite class. All sprites should inherit from this (directly or
+    indirectly).
+
+    The things that define a BaseSprite:
+
+    * The __event__ protocol (see ppb.eventlib.EventMixin)
+    * A position vector
+    * A layer
```

Comment:
Layer only makes sense on renderables

Reply:
Not really.

`Scene.sprite_layers()` exists whether you have a renderer or not.

Reply:
"Renderable" means "can be rendered", as in, sprites with rendering support.

I don't think a "layer" makes sense on Sprites that don't have any rendering support (don't have `RenderableMixin`).

Reply:
There's more use cases for layering besides rendering. A simple use case is using layers to determine collisions (as in, only things on the same layer can collide).

Reply:
In general: The `Renderer` doesn't actually care if things have layer since it leans on `Scene.sprite_layers` for render order. The `Scene` does care, because it's the one that implements iteration in layer order. `Scene` doesn't care if things are `Renderable`, so layer needs to either go on `BaseSprite` or a separate mixin, and I . . . am cautious about making one attribute mixins.

Reply:
:thinking: 

I'm skeptical about just making up use cases that abuse semantics, but I'll call the argument "good enough".